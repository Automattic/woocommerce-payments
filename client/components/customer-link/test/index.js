/** @format */
/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import CustomerLink from '../';

describe( 'CustomerLink', () => {
	test( 'renders a link to a customer with name and email', () => {
		const { container: customerLink } = renderCustomer( {
			name: 'Some Name',
			email: 'some@email.com',
		} );
		expect( customerLink ).toMatchSnapshot();
	} );

	test( 'renders a dash if customer name is undefined', () => {
		const { container: customerLink1 } = renderCustomer( null );
		expect( customerLink1 ).toMatchSnapshot();

		const { container: customerLink2 } = renderCustomer( {} );
		expect( customerLink2 ).toMatchSnapshot();
	} );

	function renderCustomer( customer ) {
		return render( <CustomerLink customer={ customer } /> );
	}
} );
