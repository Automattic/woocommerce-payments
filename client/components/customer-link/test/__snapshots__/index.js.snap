// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerLink renders a dash if customer name is undefined 1`] = `
<div>
  -
</div>
`;

exports[`CustomerLink renders a dash if customer name is undefined 2`] = `
<div>
  -
</div>
`;

exports[`CustomerLink renders a link to a customer with name and email 1`] = `
<div>
  <a
    data-link-type="wc-admin"
    href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&search%5B0%5D=Some%20Name%20(some%40email.com)"
  >
    Some Name
  </a>
</div>
`;
